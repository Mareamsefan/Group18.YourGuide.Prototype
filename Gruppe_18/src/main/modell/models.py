from sqlalchemy import Column, String, Integer, DATETIME
import uuid
from Gruppe_18.src.main.database.sql_alchemy import Base


class Account(Base):
    __tablename__ = "User"
    account_id = Column("id", String, primary_key=True)
    username = Column("username", String)
    password = Column("password", String)
    phoneNumber = Column("phoneNumber", String)
    emailAddress = Column("emailAddress", String)

    def __init__(self, username, password, phoneNumber, emailAddress):
        self.account_id = str(uuid.uuid4())
        self.username = username
        self.password = password
        self.phoneNumber = phoneNumber
        self.emailAddress = emailAddress

    def __repr__(self):
        return f"({self.account_id}) {self.username} {self.password} {self.phoneNumber} {self.emailAddress}"

    def get_id(self):
        return self.id


class Tour(Base):
    # duration in hours
    # cost in dollars
    __tablename__ = "Tour"
    tour_id = Column("id", String, primary_key=True)
    title = Column("title", String)
    date = Column("date", DATETIME)
    destination = Column("destination", String)
    duration = Column("duration", Integer)
    cost = Column("cost", Integer)
    max_travelers = Column("max_travelers", Integer)
    language = Column("language", String)
    pictureURL = Column("pictureURL", String)
    booked = Column("booked", Integer, default=0)

    def __init__(self, title, date, destination, duration, cost, max_travelers, language, pictureURL):
        self.tour_id = str(uuid.uuid4())
        self.title = title
        self.date = date
        self.destination = destination
        self.duration = duration
        self.cost = cost
        self.language = language
        self.max_travelers = max_travelers
        self.pictureURL = pictureURL

    def __repr__(self):
        return f"({self.tour_id}) {self.title} {self.destination} {self.duration} {self.cost} {self.language} {self.max_travelers} {self.pictureURL} {self.booked}"

