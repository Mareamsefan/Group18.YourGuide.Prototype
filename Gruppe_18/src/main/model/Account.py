import json
import uuid


class Account:

    def __init__(self, username, password, phoneNumber, emailAddress):
        self.id = str(uuid.uuid4())
        self.username = username
        self.password = password
        self.phoneNumber = phoneNumber
        self.emailAddress = emailAddress



    def get_id(self):
        return self.id
