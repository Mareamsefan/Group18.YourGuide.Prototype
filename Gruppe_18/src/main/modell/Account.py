import json
import uuid


class Account:

    def __init__(self, username, password, phoneNumber, emailAddress):
        self.username = username
        self.password = password
        self.phoneNumber = phoneNumber
        self.emailAddress = emailAddress
        self.account_id = str(uuid.uuid4())
        self.registered_tours = []

    def register_to_tour(self, tour):
        if tour.booked < tour.max_travelers:
            tour.book_tour()
            self.registered_tours.append(tour)
        else:
            return "you can't register to this tour"
