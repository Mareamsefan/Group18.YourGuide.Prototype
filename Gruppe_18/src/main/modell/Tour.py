import json
import uuid


class Tour:
    # duration in hours
    # cost in dollars

    def __init__(self, title, destination, duration, cost, pictureURL, language="English", max_travelers=15):
        self.pictureURL = pictureURL
        self.language = language
        self.destination = destination
        self.title = title
        self.duration = duration
        self.cost = cost
        self.max_travelers = max_travelers
        self.booked = 0
        self.tour_Id = str(uuid.uuid4())

    def book_tour(self):
        if not self.booked >= self.max_travelers:
            self.booked += 1
            return True
        else:
            return False

    def get_tour_description(self):
        return f"This tour will take you to {self.destination} for {self.duration} hours, and is " \
               f"offered in {self.language}"


