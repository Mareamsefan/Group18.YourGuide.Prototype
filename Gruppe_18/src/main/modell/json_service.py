import json


def save_to_json(entity, filename):
    try:
        with open(filename, "r") as json_file:
            filedata = json.load(json_file)
    except FileNotFoundError:
        filedata = []

    data = entity.to_dict()
    filedata.append(data)

    with open(filename, "w") as json_file:
        json.dump(filedata, json_file, indent=4)


def save_to_stream(entity, io_stream):
    data = entity.to_dict()
    try:
        filedata = json.load(io_stream)
    except json.JSONDecodeError:
        filedata = []

    filedata.append(data)

    io_stream.truncate(0)
    io_stream.seek(0)
    json.dump(filedata, io_stream, indent=4)


def to_dict(entity):
    pass


def from_dict(entity, data):
    pass
