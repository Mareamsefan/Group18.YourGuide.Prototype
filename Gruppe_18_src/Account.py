import json


class Account:

    def __init__(self, username, password, phoneNumber, emailAddress):
        self.username = username
        self.password = password
        self.phoneNumber = phoneNumber
        self.emailAddress = emailAddress

    def successful_registration(self):
        self.save_to_json()
        return True

    def delete_account(self, username):
        with open("account.json", "r") as json_file:
            data = json.load(json_file)
            if username in data:
                del data[username]

                with open("account.json_2", "w") as json_file:
                    json.dump(data, json_file, indent=4)
                return True
            else:
                return False

    def save_to_json(self):
        account_data = {
            "username": self.username,
            "password": self.password,
            "phoneNumber": self.phoneNumber,
            "emailAddress": self.emailAddress

        }

        with open("account.json", "w") as json_file:
            json.dump(account_data, json_file, indent=4)
