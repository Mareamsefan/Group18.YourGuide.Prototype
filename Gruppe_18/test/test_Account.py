import pytest
from approvaltests import verify, Options
from approvaltests.scrubbers import scrub_all_guids
from Gruppe_18.src.main.modell.Account import Account
from io import StringIO

approval_options = Options().with_scrubber(scrub_all_guids)

@pytest.fixture
def account():
    return Account(
        "username",
        "password",
        12345678,
        "user_@gmail.com"
    )

def test_account_can_be_created_and_saved(account):
    io_stream = StringIO()
    account.save_to_stream(io_stream)
    saved_data = io_stream.getvalue()
    io_stream.seek(0)

    verify(saved_data, options=approval_options)

def test_account_had_an_successful_registration(account):
    assert account.successful_registration() == True

def test_user_can_delete_their_account(account):
    assert account.delete_account() == True

