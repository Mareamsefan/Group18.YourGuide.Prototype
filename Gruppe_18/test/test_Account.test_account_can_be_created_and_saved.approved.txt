[
    {
        "id": "<guid_0>",
        "username": "username",
        "password": "password",
        "phoneNumber": 12345678,
        "emailAddress": "user_@gmail.com"
    }
]
